packages/typir/src/utils/utils-type-comparison.ts

```diff
@@ -215,46 +215,101 @@ export function compareTypes(leftTypes: Array<Type | undefined>, rightTypes: Arr
     const targetCopy = new Map(targetFields);
     const conflicts: IndexedTypeConflict[] = [];
     for (const entry of sourceFields.entries()) {
         const sourceType = entry[1];
         const name = entry[0];
         if (targetCopy.has(name)) {
             // field exists in both maps
-            const targetType = targetCopy.get(name)!;
+            const targetType = targetCopy.get(name);
             targetCopy.delete(name);
-            const comparisonResult = comparator(sourceType, targetType);
-            if (comparisonResult !== true) {
-                // different types
+            if (sourceType === undefined && targetType === undefined) {
+                // both types don't exist, this is OK
+            } else if (sourceType === undefined && targetType !== undefined) {
+                // only the target type exists
                 conflicts.push({
-                    expected: sourceType,
+                    expected: undefined,
                     actual: targetType,
                     index: name,
-                    subProblems: [comparisonResult]
+                    subProblems: []
+                });
+            } else if (sourceType !== undefined && targetType === undefined) {
+                // only the source type exists
+                conflicts.push({
+                    expected: sourceType,
+                    actual: undefined,
+                    index: name,
+                    subProblems: []
                 });
+            } else if (sourceType !== undefined && targetType !== undefined) {
+                // both types exist => compare them
+                const comparisonResult = comparator(sourceType, targetType);
+                if (comparisonResult !== true) {
+                    // different types
+                    conflicts.push({
+                        expected: sourceType,
+                        actual: targetType,
+                        index: name,
+                        subProblems: [comparisonResult]
+                    });
+                } else {
+                    // same type
+                }
             } else {
-                // same type
+                throw new Error('impossible case');
             }
         } else {
             // field is missing in target
-            conflicts.push({
-                expected: sourceType,
-                actual: undefined,
-                index: name,
-                subProblems: []
-            });
+            if (sourceType === undefined) {
+                // this is OK
+            } else {
+                conflicts.push({
+                    expected: sourceType,
+                    actual: undefined,
+                    index: name,
+                    subProblems: []
+                });
+            }
         }
     }
     // fields are missing in source
     for (const entry of targetCopy.entries()) {
```

Comment:
`for (const [index, actual] of targetCopy.entries()) {`

Reply:
fixed